upath/implementations/azure.py

```diff
@@ -0,0 +1,27 @@
+from .. import core
+import os
+import re
+
+
+class _AzureAccessor(core._FSSpecAccessor):
+    def __init__(self, parsed_url, *args, **kwargs):
+        super().__init__(parsed_url, *args, **kwargs)
+
+    def _format_path(self, s):
+        """If the filesystem backend doesn't have a root_marker, strip the
+        leading slash of a path and add the bucket
+        """
+        return os.path.join(self._url.netloc, s.lstrip("/"))
+
+
+class AzurePath(core.UPath):
+    _default_accessor = _AzureAccessor
+
+    def _init(self, *args, template=None, **kwargs):
+        if kwargs.get("storage_options"):
+            kwargs.update(kwargs["storage_options"])
```

Comment:
I'm curious if this is why the pickling tests are failing? I think it is possible that the pickled kwargs will not match the _init kwargs, leading to an inconsistent load state.